### ISSUE TITLE
Split Script.js into multiple files

### ISSUE THREAD

**Issue by zamhaq** [fossasia/susi_chromebot#156]
**I'm submitting a ...**  (check one with "x")
- [ ] bug report
- [x] feature request
- [ ] support request => Please do not submit support requests here, instead ask your query in out Gitter channel at https://gitter.im/fossasia/susi_server

**Current behavior:**
The extension loads from script.js which results in lots of conflicts when multiple members edit the same file. Also, the codebase is increasing and its reducing readability being in a single file.

**Expected behavior:**
Split script.js into different modules and bundle them using `webpack`. Then load the extension from the
bundled file.
 
**Would you like to work on the issue?**
Yes.

**Comment by zamhaq** [comment]
@gabru-md @mariobehling @ms10398 what do you think?

**Comment by ms10398** [comment]
Yes thats a good idea

**Comment by gabru-md** [comment]
Yes it definitely is a good idea. I was thinking of proposing it as well.
:+1:

**Comment by akshatnitd** [comment]
@zamhaq Are you working on it ? I am free to work on the issue

**Comment by zamhaq** [comment]
@akshatnitd yes I am. Please look into this issue #149 and leave a comment there if you would like to work on that. 😀

**Comment by zamhaq** [comment]
@gabru-md @ms10398 I think the time for this refactoring is too high. Changing the structure while current PR's are incoming is also difficult since, I would have to manually go and change those things in the split files while I am working on this or I will have to block other PR's first and then later when the new structure is uploaded then tell the PR makers to make the necessary changes. Both ways it stagnates the growth of the project. Also, I don't have much time because of exams. Should we drop this?

**Comment by gabru-md** [comment]
@zamhaq It is okay!
You can focus on the exams for now. I think that with the current rate of development it will be difficult for us to split one script into multiple files as well. But I suppose that this idea must not be dropped but can be **kept on a hold** while we undergo development.
Cheers!
All the best for your exams!

**Comment by zamhaq** [comment]
Ok then, after my exams I will continue my work, and maybe we can collaborate on splitting the file.

**Comment by ms10398** [comment]
Yes sure @zamhaq we can go for it after a while surely :+1:

**Comment by ms10398** [comment]
@gabru-md @zamhaq I am putting it as a Google code-in task for the Google code-in students.

**Comment by acamit** [comment]
@ms10398 @zamhaq  I would like to work on this issue. Is anyone working on it currently?

**Comment by ms10398** [comment]
Are you a GCI participant

**Comment by acamit** [comment]
No, i am not gci participant.

Thanks and Regards
Amit Chawla

**Comment by stealthanthrax** [comment]
@acamit , this task is reserved for GCI participants ,you can browse other tasks on this repo 😄

**Comment by gabru-md** [comment]
@zamhaq I will start working on this now. 
Since I have added all response So i guess now we can go ahead and split the files.
Meanwhile you can aa prettier and eslint as well.
:)

**Comment by gabru-md** [comment]
@zamhaq @ms10398 Since I am working on this now and I have to design a workflow and then split the files, Can you guys please ensure that no PRs are merged, So that no conflicts are caused by this ?

**Comment by ms10398** [comment]
@gabru-md , Yes then take it as a **priority** issue. And make the PR within one to two days.

Thanks!

**Comment by zamhaq** [comment]
@gabru-md @ms10398 any plans on this one? Should we do it now since GCI is coming up, it will help new contributors to get started easily. What say? :smiley:

**Comment by gabru-md** [comment]
sure it will.
I guess we can have simple tasks for GCI students and have codeheat students take this up @zamhaq ?

**Comment by rupav** [comment]
Would like to do the initial setup with webpack. 👍
